### ISSUE TITLE
Nav: Refactor - JSX/TSX Good First Issue

### ISSUE THREAD

**Issue by Medic1111** [Medic1111/FINALLY-SEEN#77]
Hey guys... getting there.
The Nav right now is a hot mess. We have 6 components in total Nav related. 
Here are things to consider: 

- [ ] The FOOTER nav will always display the content of a Logged out Nav
- [ ] The Hamburger Menu content will always match the content of the current Nav (logged in or logged out)
- [ ] The state responsible for checking whether a user is logged in the in the FEATURE/ AUTH CTX/ isAuth (currently set as false)
- [ ] The buttons BUTTON BOX component should still be present as it currently is

--- 

The Challenge:

- Study the current structure and components and find a way to bring compose the functionality in no more than TWO components, maintainability must be present, and Footer nav doesn't count as a component (but could be if wanted)

If you feel like you can tackle that task, please comment below to be assigned to it. Dont forget to read the contributing.md and Open PRs to Development Branch.

Thank you and Happy Collab =)

**Comment by coderhersh** [comment]
Hi, there!! Can I Work on this issue??

**Comment by Medic1111** [comment]
@coderhersh Its yours! Have fun. If anything, feel free to ask questions here in the comments =)

**Comment by Dismated** [comment]
Hi, is the issue still open? Can I take it?

**Comment by Medic1111** [comment]
@Dismated Absolutely. I see our friend ended up not being able to contribute so Ill make it yours =) 
Have Fun!

**Comment by Medic1111** [comment]
@Dismated any progress? I just pushed some commits Nav related. Let me know. If you are no longer able to collab please let me know so I can reassign =) Thanks

**Comment by JoshiJaimin** [comment]
Is this issue is still available to contribute?
I would like to work in this issue.

**Comment by Medic1111** [comment]
@JoshiJaimin Ill transfer this issue to you =) 
Thank you so much!

**Comment by Sarthak00007** [comment]
@Medic1111 

Is this issue still available to contribute?
I would like to work on this issue.

**Comment by Duggi-Niharika-Sai-06** [comment]
I need to work on open source projects as a part of my university course. I am interested to contribute to this problem.

Can you please assign this to me?

**Comment by imen22bnas** [comment]
Hi! I'd like to work on this issue. Can I take it?

**Comment by hrajput0322** [comment]
do these buttons even work? The register and login buttons in the navbar

**Comment by DipakHalkude** [comment]
I’ve refactored the navigation system as discussed in the open issue. Here’s what I’ve done:

Reduced the 6 Nav-related components to 2 main components: Nav and HamburgerMenu.

Ensured the Footer nav remains logged-out only.

Hamburger menu now dynamically matches the current isAuth state.

Preserved the ButtonBox component in both Nav and Hamburger.

Maintained clean and maintainable structure for easier future updates.

I’ve tested the changes locally and everything is working as expected. Please review the PR and let me know if any improvements are needed.

Thank you for considering my contribution!

**Comment by shreyasbp317** [comment]
can i contribute to this issue?

**Comment by mdcode-00** [comment]
Hi! I’m a beginner MERN developer. 
I would like to work on this issue. 
Can you please assign it to me?

**Comment by nageshkumar74** [comment]
Hi! I’m a MERN developer.
I would like to work on this issue.
Can you please assign it to me

**Comment by Bhaveshkransan** [comment]
Hi, there!! Can I Work on this issue??
